warehouse/query-core/src/main/java/org/apache/commons/jexl2/parser/JexlNodes.java

```diff
@@ -78,6 +79,20 @@ public static <T extends JexlNode> T children(T node, JexlNode... children) {
         return node;
     }
     
+    /**
+     * Sets the supplied child array as the children member of {node} and sets the parent reference of each element in {children} to {node}.
+     *
+     * @param node
+     * @param children
+     * @return
+     */
+    public static <T extends JexlNode> T children(T node, Collection<JexlNode> children) {
+        node.children = children.toArray(new JexlNode[0]);
```

Comment:
You can delegate to the existing JexlNode.children() method by shortening this to 
```java
return children(node, children.toArray(new JexlNode[0]));
```
and thereby avoid repeating some code.

Reply:
Updated new children method to create a children array and pass it to the existing children method.